examples/registration_form/lib/address/address_component.dart

```diff
@@ -0,0 +1,110 @@
+import 'package:angular/angular.dart';
+import 'package:angular_components/angular_components.dart';
+import 'package:angular_forms/angular_forms.dart';
+
+@Component(
+    selector: 'address-cmp',
+    templateUrl: 'address_component.html',
+    exports: [
+      states
+    ],
+    directives: [
+      formDirectives,
+      materialInputDirectives,
+      MaterialAutoSuggestInputComponent,
+      RequiredFirstValidator,
+      RequiredState
+    ])
+class AddressComponent {
+  String address1;
+  String address2;
+  String city;
+  String state;
+  String zip;
+}
+
+/// A [Validator] which validates that the [requiredFirst] value is preset when
+/// this control has a value set.
```

Comment:
This comment didn't help me too much understanding this validator. Maybe:
Validator that ensures that [requiredFirst] control has a value before this control. For example an address input where given two address lines the second line shouldn't be the only one with a value.

Reply:
I definitely struggled a bit with writing this one. Thanks for the suggestion, much better than what I originally came up with.